docs/source/developers/release.rst

```diff
@@ -216,123 +216,365 @@ Be sure to go through on the following checklist:
 #. Start the new version on JIRA on the ARROW project
 #. Start the new version on JIRA for the related CPP PARQUET version
 #. Merge changes on release branch to maintenance branch for patch releases
+#. Add the new release to the Apache Reporter System
 #. Upload source
 #. Upload binaries
 #. Update website
 #. Update Homebrew packages
 #. Update MSYS2 package
 #. Upload RubyGems
-#. Upload JS packages
+#. Upload JavaScript packages
 #. Upload C# packages
 #. Update conda recipes
 #. Upload wheels/sdist to pypi
 #. Publish Maven artifacts
 #. Update R packages
 #. Update vcpkg port
+#. Update Conan recipe
 #. Bump versions
 #. Update tags for Go modules
 #. Update docs
+#. Announce the new release
 #. Remove old artifacts
```

Comment:
Should we add the update of the Arrow version on the cookbooks as a post-release task too? The main problem is that is requires the conda packages to be available at the moment and they might take a while. I am monitoring for them to be available for 10.0.0 to update the cookbooks version at the moment.

Reply:
Wow, I didn't know that!
 I've added a place holder for it. Could you fill it in a follow-up pull request?